### PR TITLE
site: Update annotations doc with better ingress.class information

### PR DESCRIPTION
**Pull request by youngnick**
Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: site/docs/master/annotations.md [hunk 1]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
 
 The following Kubernetes annotations are supported on [`Ingress`] objects:
 
- - `kubernetes.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. If not set, then all Ingress controllers serve the Ingress. If specified as `kubernetes.io/ingress.class: contour`, then Contour serves the Ingress. If any other value, Contour ignores the Ingress definition. You can override the default class `contour` with the `--ingress-class-name` flag at runtime. This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+### ingress.class
```
Comment (jpeach) [inline site/docs/master/annotations.md:16]: This is better as "Ingress class".

File: site/docs/master/annotations.md [hunk 2]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
 
 The following Kubernetes annotations are supported on [`Ingress`] objects:
```
Comment (jpeach) [inline site/docs/master/annotations.md:14]: Why are there square brackets here?

File: site/docs/master/annotations.md [hunk 3]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
 
- - `kubernetes.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. If not set, then all Ingress controllers serve the Ingress. If specified as `kubernetes.io/ingress.class: contour`, then Contour serves the Ingress. If any other value, Contour ignores the Ingress definition. You can override the default class `contour` with the `--ingress-class-name` flag at runtime. This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+### ingress.class
+
+The Ingress class that should interpret and serve the Ingress.
```
Comment (jpeach) [inline site/docs/master/annotations.md:18]: How about:

```
The Ingress class annotation can be used to specify which Ingress controller should serve a particular Ingress object.
```

And just join this up with the subsequent paragraph.

File: site/docs/master/annotations.md [hunk 4]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
+In both cases, they will behave as follows, by default:
+
+* If not set, then all Ingress controllers serve the Ingress.
+* If specified as `kubernetes.io/ingress.class: contour`, then Contour serves the Ingress.
+* If any other value, Contour ignores the Ingress definition.
+
+You can override the default class `contour` with the `--ingress-class-name` flag at runtime.
+This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+
+* If you do this, the behavior is as follows:
```
Comment (jpeach) [inline site/docs/master/annotations.md:30]: Make this line part of the preceding paragraph, not a bullet point.

File: site/docs/master/annotations.md [hunk 5]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
+* If the annotation is not set, Contour will ignore the ingress.
+* If the annotation is set to any value other than what you passed to the `--ingress-class-name` flag, Contour will ignore the ingress.
+* If the annotation matches the value that you passed to `--ingress-class-name` flag, Contour will serve the Ingress.
+
+This same logic applies for these annotations on HTTPProxy, and IngressRoute (deprecated) objects.
```
Comment (jpeach) [inline site/docs/master/annotations.md:35]: No need for the comma here.

File: site/docs/master/annotations.md [hunk 6]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
+The Ingress class that should interpret and serve the Ingress.
+
+This annotation may be specified as the standard `kubernetes.io/ingress.class` or a Contour-specific `projectcontour.io/ingress.class`.
+In both cases, they will behave as follows, by default:
+
+* If not set, then all Ingress controllers serve the Ingress.
+* If specified as `kubernetes.io/ingress.class: contour`, then Contour serves the Ingress.
+* If any other value, Contour ignores the Ingress definition.
+
+You can override the default class `contour` with the `--ingress-class-name` flag at runtime.
```
Comment (jpeach) [inline site/docs/master/annotations.md:27]: Maybe just

```
by providing the `--ingress-class-name` flag to Contour. 
```

File: site/docs/master/annotations.md [hunk 7]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
+This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+
+* If you do this, the behavior is as follows:
+* If the annotation is not set, Contour will ignore the ingress.
+* If the annotation is set to any value other than what you passed to the `--ingress-class-name` flag, Contour will ignore the ingress.
```
Comment (jpeach) [inline site/docs/master/annotations.md:32]: "other than the one passed"

Comment (jpeach) [reply site/docs/master/annotations.md:32]: Should consistently capitalize `Ingress` here and on the line above.

File: site/docs/master/annotations.md [hunk 8]
```diff
@@ -13,15 +13,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
 
 The `ingress.kubernetes.io/force-ssl-redirect` annotation takes precedence over `kubernetes.io/ingress.allow-http`. If they are set to `"true"` and `"false"` respectively, Contour *will* create an Envoy HTTP route for the Virtual host, and set the `require_tls` virtual host option.
 
 ## Contour specific Ingress annotations
 
- - `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. If not set, then all Ingress controllers serve the Ingress. If specified as `projectcontour.io/ingress.class: contour`, then Contour serves the Ingress. If any other value, Contour ignores the Ingress definition. You can override the default class `contour` with the `--ingress-class-name` flag at runtime. This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+ - `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. See the [main ingress-class annotation section](#ingress.class) for more details.
```
Comment (jpeach) [inline site/docs/master/annotations.md:46]: s/ingress-class/Ingress class/ (and update anchor target)

File: site/docs/master/annotations.md [hunk 9]
```diff
@@ -41,8 +63,7 @@ A [Kubernetes Service][9] maps to an [Envoy Cluster][10]. Envoy clusters have ma
   - The `tls` protocol allows for requests which terminate at Envoy to proxy via tls to the upstream. _Note: This does not validate the upstream certificate._
 
 ## Contour specific HTTPProxy annotations
-- `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the HTTPProxy. If not set, then all all Contour instances serve the HTTPProxy. If specified as `projectcontour.io/ingress.class: contour`, then Contour serves the HTTPProxy and any others that have no annotation defined. If any other value, Contour ignores the HTTPProxy definition.
-You can override the default class `contour` with the `--ingress-class-name` flag at runtime.
+- `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the HTTPProxy.See the [main ingress-class annotation section](#ingress.class) for more details.
```
Comment (jpeach) [inline site/docs/master/annotations.md:66]: s/ingress-class/Ingress class/ (and update anchor target)

File: site/docs/master/annotations.md (hunk 2 above)
Comment (youngnick) [reply site/docs/master/annotations.md:14]: This one was inherited, no idea, I'll remove.

File: site/docs/master/annotations.md (hunk 1 above)
Comment (youngnick) [reply site/docs/master/annotations.md:16]: Sure.

File: site/docs/master/annotations.md [hunk 10]
```diff
@@ -41,8 +61,7 @@ A [Kubernetes Service][9] maps to an [Envoy Cluster][10]. Envoy clusters have ma
   - The `tls` protocol allows for requests which terminate at Envoy to proxy via tls to the upstream. _Note: This does not validate the upstream certificate._
 
 ## Contour specific HTTPProxy annotations
-- `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the HTTPProxy. If not set, then all all Contour instances serve the HTTPProxy. If specified as `projectcontour.io/ingress.class: contour`, then Contour serves the HTTPProxy and any others that have no annotation defined. If any other value, Contour ignores the HTTPProxy definition.
-You can override the default class `contour` with the `--ingress-class-name` flag at runtime.
+- `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the HTTPProxy.See the [main ingress-class annotation section](#ingress-class) for more details.
```
Comment (jpeach) [inline site/docs/master/annotations.md:64]: - space after the `.`
- "main Ingress class annotation section"

File: site/docs/master/annotations.md [hunk 11]
```diff
@@ -11,17 +11,37 @@ However, Contour still supports a number of annotations on the Ingress resources
...
 
 The `ingress.kubernetes.io/force-ssl-redirect` annotation takes precedence over `kubernetes.io/ingress.allow-http`. If they are set to `"true"` and `"false"` respectively, Contour *will* create an Envoy HTTP route for the Virtual host, and set the `require_tls` virtual host option.
 
 ## Contour specific Ingress annotations
 
- - `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. If not set, then all Ingress controllers serve the Ingress. If specified as `projectcontour.io/ingress.class: contour`, then Contour serves the Ingress. If any other value, Contour ignores the Ingress definition. You can override the default class `contour` with the `--ingress-class-name` flag at runtime. This can be useful while you are migrating from another controller, or if you need multiple instances of Contour.
+ - `projectcontour.io/ingress.class`: The Ingress class that should interpret and serve the Ingress. See the [main ingress-class annotation section](#ingress-class) for more details.
```
Comment (jpeach) [inline site/docs/master/annotations.md:44]: "main Ingress class annotation section"

**Pull request merged**